TESTING.md

```diff
@@ -1,70 +1,88 @@
-1. Install the Datadog Plugin, either via the Update Center, or via .hpi
-2. Configure with an API key
-3. Create a test build
-
-Now, we need to connect to the squid docker container to do some introspection, so we can prove that the proxy is routing this data.
-
-    docker exec -it squid bash
-
-Once you are conencted to the squid box, run a tcp dump to follow port 3128.
-
-    apt-get update
-    apt-get install -y tcpdump
-    sudo tcpdump -i eth0 -vvvvtttAXns 1500 'port 3128'
-
-Now that this window is following all the traffic coming through port 3128, let's go back to the Jenkins UI.
-
-1. Run the test build that you had setup previously.
-2. Look on the output from the tcpdump. You should NOT see any output.
-3. Now go to `Manage Jenkins` > `Manage Plugins`, and then select the `Advanced` tab.
-4. From here you will see the 'HTTP Proxy Configuration' section. Enter the following information:
-    * Server: 172.17.42.1
-    * Port: 3128
-5. Optionally, you can click `Advanced`, and then test `http://www.google.com` and you should see the output in the tcpdump terminal window.
-6. Finally, run the test build again.
-7. You should now see output from the tcpdump terminal window!
-
-Repeat this by enabling/disabling the proxy configuration in Jenkins and repeating the test to prove to yourself that it was not a fluke.
-**Note: I've noticed that Jenkins seems to send a disconnection series of packets to the proxy server with the first connection attempt (when you run your test build), after you've disabled the proxy configration. Run the test job a second time, and you'll again not see any output.**
+    ```
+    version: "3.7"
+    services:
+      jenkins:
+        image: jenkins/jenkins:lts
+        ports:
+          - 8080:8080
+        volumes:
+          - $JENKINS_PLUGIN/target/:/var/jenkins_home/plugins
+      
+      dogstatsd:    
+        image: datadog/dogstatsd:latest 
+        environment:
+          - DD_API_KEY=<API_KEY>
+          - DD_LOG_LEVEL=debug
+        ports:
+          - 8125:8125/udp  
+               
+    ```
 
-## Check Style
-In order to check that the Java style meets the recommendations of Sun and Google, as closely as possible, here is a way to test it.
+2. Set the `JENKINS_PLUGIN` environment variable to point to the folder of the clone/fork of this repository.
+3. Run `docker-compose -f <DOCKER_COMPOSE_FILE_PATH> up`.
+4. Check your terminal and look for the admin password:
+    ```
+    jenkins_1    | *************************************************************
+    jenkins_1    | *************************************************************
+    jenkins_1    | *************************************************************
+    jenkins_1    |
+    jenkins_1    | Jenkins initial setup is required. An admin user has been created and a password generated.
+    jenkins_1    | Please use the following password to proceed to installation:
+    jenkins_1    |
+    jenkins_1    | 70dfe3d2b23841cc96c101c0521716f0
```

Comment:
I know this pw isn't in use but maybe something like the following so its clearer:
```suggestion
    jenkins_1    | <JENKINS_ADMIN_PASSWORD>
```